vgen/src/test/java/io/gapi/vgen/testdata/go_library.baseline

```diff
@@ -191,14 +189,14 @@ func (book Book) Path() (string, error) {
 
 // CreateShelf creates a shelf, and returns the new Shelf.
 func (client *Client) CreateShelf(ctx context.Context, req *google_example_library_v1.CreateShelfRequest, opts ...gax.CallOption) (*google_example_library_v1.Shelf, error) {
-	callOpts := append(nonIdempotentRetryCodes(), defaultRetryParams(), client.CallOptions["CreateShelf"]...)
-	callOpts = append(callOpts, opts...)
+	defaultCallOpts := append(nonIdempotentRetryCodes(), defaultRetryParams()...)
+	callOpts := append(client.CallOptions["CreateShelf"], opts...)
```

Comment:
I am a little worried about this. If `client.CallOptions[s]` is not a filled slice, we will have a race condition if two goroutines call the same method on the same client concurrently. See example in https://play.golang.org/p/tgObgdGEsG .

It probably also worth documenting that the user may not assign/modify client.CallOptions concurrently with any calls on the client.